From 6e4b8bb524294932929569d51e3b49c19cbb2844 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@st.com>
Date: Fri, 21 Feb 2025 12:55:37 +0100
Subject: [PATCH 1553/1695] product: tfm-stm32mp2: tfm_clock: check context
 NULLity

Outlined by coverity

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 60b248549c4d1c475667950445c95890dfc58ed0 product: tfm-stm32mp2: tfm_clock: check context NULLity]
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
Change-Id: I012ed3c0ea06ee664a4b3a5ee7925d9b50615882
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/408745
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
index 55391c8e8..3c5d1761c 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
@@ -415,6 +415,10 @@ static int tfm_clock_element_init(fwk_id_t element_id, unsigned int dev_count,
     const struct mod_tfm_clock_config *config = data;
     int res;
 
+    if (ctx == NULL) {
+        return FWK_E_PARAM;
+    }
+
     ctx->clk = (struct clk *)config->clk;
     if (ctx->clk) {
         ctx->enabled = config->default_enabled;
-- 
2.47.2

